### PR TITLE
[redfish] Add Reset and BootImage actions

### PR DESCRIPTION
**Pull request by zyoung51**
- Add #Computer.Reset action to /Systems
- Add #RackHD.BootImage action to /Systems
- Refactor redfish validator to read schemaDirs from config
  and added Oem schema to redfish v1
- Refactor systems to use node-service-api and makeOptions

### CODE REVIEW

**Comment by zyoung51** [comment]
@RackHD/corecommitters @uppalk1 @keedya @tannoa2 
and @geoff-reid @brianparry for changes to node-api-service.

**Comment by brianparry** [comment]
:+1:

**Comment by jlongever** [comment]
+1

File: lib/api/1.1/northbound/nodes.js [hunk 1]
```diff
@@ -282,7 +282,7 @@ function nodesRouterFactory (
      */
 
     router.get('/nodes/:identifier/catalogs/:source', rest(function (req) {
-        return nodeApiService.getNodeCatalogSourceById(req);
+        return nodeApiService.getNodeCatalogSourceById(req.params.identifier, req.params.source);
```
Comment (benbp) [inline lib/api/1.1/northbound/nodes.js]: Oh thank god.

File: lib/api/redfish-1.0/systems.js [hunk 2]
```diff
@@ -373,3 +374,138 @@ function getSelLogServiceEntry(req, res)  {
...
+    redfish.validate(payload, 'RackHD.ResetAction.json#/definitions/ResetAction')
+    .then(function(result) {
+        var map = {
+            On: 'Graph.PowerOn.Node',
+            ForceOff: 'Graph.PowerOff.Node',
+            GracefulRestart: 'Graph.Reset.Soft.Node',
+            ForceRestart: 'Graph.Reboot.Node',
+            ForceOn: 'Graph.PowerOn.Node',
+            PushPowerButton: 'Graph.Reset.Soft.Node'
+        }
```
Comment (benbp) [inline lib/api/redfish-1.0/systems.js]: Please add this semicolon because I'm a pedant :)

**Comment by benbp** [comment]
:+1: when you add that semicolon

**Comment by zyoung51** [comment]
test this please

**Comment by jlongever** [comment]
http://rackhdci.lss.emc.com/job/on-http/328/ passed.
+1

**Pull request merged**